lib/kitto/plugs/authentication.ex

```diff
@@ -0,0 +1,31 @@
+defmodule Kitto.Plugs.Authentication do
+  import Plug.Conn
+
+  def init(opts), do: opts
+
+  def call(conn, _) do
+    if authentication_required?(conn) && !authenticated?(conn) do
+      conn |> send_resp(401, "Authorization required") |> halt
+    else
+      conn
+    end
+  end
+
+  defp auth_token, do: Application.get_env(:kitto, :auth_token)
+
+  defp authentication_required?(conn) do
+    !!auth_token && validate_request(conn)
+  end
+
+  defp validate_request(conn) do
+    conn.method == "POST" && conn.request_path =~ ~r/^\/?widgets/
```

Comment:
Alright, so after a bit of chat on the Elixir Slack channel it would seem that my ideas for building out the API like above is a bit against the idea behind the Plug model. The general recommendation was to split routes into 2 routers: authenticated routes and unauthenticated router and forward between them. Not sure what the thoughts were on that idea.